nilearn/surface/surface.py

```diff
@@ -1010,3 +1010,114 @@ def check_surface(surface):
     mesh, data = check_mesh_and_data(surface.mesh,
                                      surface.data)
     return Surface(mesh, data)
+
+def compute_vertex_neighborhoods(surface):
+    """For each vertex, compute the neighborhood defined as all the vertices that are connected by a face.
+    
+    Parameters
+    ----------
+    surface : Surface-like (see description)
+    
+    Returns
+    -------
+    neighbors : A list of all the vertices that are connected to each vertex
+    
+    """
+    # This builds a coordiate to face dictionary 
+    # where each entry gives a list of the faces associated with that vertex
+    d = {k:[] for k in range(surface.coordinates.shape[0])}
+    for (i,(u,v,w)) in enumerate(surface.faces):
+        d[u].append(i)
+        d[v].append(i)
+        d[w].append(i)
+    
+    # This collects a list of vertices that are connected to each vertex
+    neighbors = []
+    for k in range(surface.coordinates.shape[0]):
+        neighbors.append(np.unique(surface.faces[d[k]]))
+        
+    return neighbors
+        
+def smooth_surface_data(surface, surf_data, smooth_steps = 1, method = 'sparse', weights = None):
+    """Smooth values along the surface.
+    
+    Parameters
+    ----------
+    surface : Surface-like (see description)
+
+    surf_data : Array of values at each vertex. Should be a vertics X N array. In the
+    case of fmri data, N could be number of timepoints. Each column is smoothed independently
+
+    smooth_steps : How many times to repeat the smoothing operation. Defaults to 1 step
+
+    method : Whether to use scipy sparse matrices (default)
+
+    weights : Whether to add weighting to the smoothing. Default is none
+    
+    Returns
+    -------
+    surf_data_smooth : Array of smoothed values at each vertex
+
+    Examples
+    -------
+    >>> from nilearn import datasets, surface, plotting
+    >>> fsaverage = datasets.fetch_surf_fsaverage('fsaverage')
+    >>> white_left = surface.load_surf_mesh(fsaverage.white_left)
+    >>> curv = surface.load_surf_data(fsaverage.curv_left)
+    >>> curv_smooth = surface.smooth_surface_data(surface = white_left, surf_data = curv, smooth_steps=50)
+    >>> plotting.plot_surf(white_left, surf_map = curv_smooth)
+    
+    """
+    if method == 'sparse':
```

Comment:
Why wouldn't one use a sparse method ?

Reply:
I didn't really understand the difference between the two methods (it feels to me that the `sparse` method would be faster since there are less `for` loops).